src/metabase/query_processor/streaming/xlsx.clj

```diff
@@ -290,9 +290,9 @@
 
 (defn- rounds-to-int?
   "Returns whether a number should be formatted as an integer after being rounded to 2 decimal places."
-  [value]
-  (let [rounded (.setScale (bigdec value) 2 java.math.RoundingMode/HALF_UP)]
-    (== (bigint rounded) rounded)))
+  [^double value]
+  (let [x100 (* value 100.0)]
+    (== (/ (Math/round x100) 100.0) (Math/floor value))))
```

Comment:
This doesn't seem to be quite right. I think we need accurate precision at least for small numbers like this.

```clojure
(->> [1.006 1.005 1.004 -1.006 -1.005 -1.004 1.994 1.995 1.996 -1.994 -1.995 -1.996]
     (keep (fn [v]
             (let [old (let [rounded (.setScale (bigdec v) 2 java.math.RoundingMode/HALF_UP)]
                         (== (bigint rounded) rounded))
                   new (let [x100 (* v 100.0)]
                         (== (/ (Math/round x100) 100.0) (Math/floor v)))]
               (when (not= old new)
                 [v old new])))))

;; [-1.004 true false]
;; [1.995  true false]
;; [1.996  true false]
;; [-1.995 true false]
```

Reply:
I see you called this out in the description 🤦 

I strongly suspect customers will be surprised by this formatting change. I also suspect there is another way to do this efficiently, perhaps using mod?

Reply:
I tried out a whole bunch of variants of both float arithmetic and direct operations on the underlying bits, and there were always some cases I couldn't get right.

We can at least save conversion to a BigInt I think though:

```clojure
(defn rounds-to-int? [v]
  (let [bd     (BigDecimal/valueOf ^double v)
        scaled (.setScale bd 2 RoundingMode/HALF_UP)]
    (zero? (.scale (.stripTrailingZeros scaled)))))
```

Reply:
Some optimizations I'm not sure would be worth it.

```clojure
(defn integer-when-rounded? [^double v]
  (or (= v (Math/round v))
      (let [bd (BigDecimal/valueOf v)]
        (or (< (.scale bd) 3)
            (not (pos? (-> bd (.setScale 2 RoundingMode/HALF_UP) .stripTrailingZeros .scale)))))))
```